### ISSUE TITLE
Incorrect defaultValue with nested resource attributes in SimpleForm input components

### ISSUE THREAD

**Issue by cytomich** [marmelab/admin-on-rest#384]
Sometimes need to send to server some nested attributes.
For this case, I find some strange behavior `SelectInput` and its `defaultValue` prop.

If `SelectInput` ' s `defaultValue` prop is set serialization of form data works incorrect and the component does not show its default value.

Example:

When I use component like this
```js
<SelectInput
    label="Some nested field"
    source="foo_bar"
    choices={[
        { id: 'foo1', name: 'bar1' },
        { id: 'foo2', name: 'bar2' },
    ]}
    defaultValue="foo1"
/>
```
Everything is ok
<img width="330" alt="4" src="https://cloud.githubusercontent.com/assets/3443470/23546397/69b7e12c-0010-11e7-945e-7e10a4534b98.png">

When I try set nested source attribute
```js
<SelectInput
    label="Some nested field"
    source="foo.bar" // <--
    choices={[
        { id: 'foo1', name: 'bar1' },
        { id: 'foo2', name: 'bar2' },
    ]}
    defaultValue="foo1"
/>
```

1. Its look like this
<img width="312" alt="1" src="https://cloud.githubusercontent.com/assets/3443470/23546337/31ea8eca-0010-11e7-8408-9babd83d6702.png">

2. The most interesting thing happening when I try to save resource with nested property

<img width="727" alt="2" src="https://cloud.githubusercontent.com/assets/3443470/23546509/e784a9be-0010-11e7-8718-3acf99f36515.png">

If I remove `defaultValue` property from a component.

```js
<SelectInput
    label="Some nested field"
    source="foo.bar"
    choices={[
        { id: 'foo1', name: 'bar1' },
        { id: 'foo2', name: 'bar2' },
    ]}
/>
```
Serialization works correctly.

<img width="730" alt="3" src="https://cloud.githubusercontent.com/assets/3443470/23546551/22365b20-0011-11e7-8d89-917ffdcdfad5.png">

**Comment by cytomich** [comment]
Now I have discovered that the issue with any input component, not only with `SelectInput `.

`TextField` for example:

```js
<TextInput label="Some nested field" source="foo.bar" defaultValue="Some text" />
```

looks like this

![image](https://cloud.githubusercontent.com/assets/3443470/23550085/4fd029bc-0020-11e7-9259-8ff6d85bff5d.png)

**Comment by cytomich** [comment]
Fix here #388
